scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -2219,36 +2259,44 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
     val (isValParam, isVarParam) = (ownerIsType && token.is[`val`], ownerIsType && token.is[`var`])
     if (isValParam) { mods :+= atPos(in.tokenPos, in.tokenPos)(Mod.ValParam()); next() }
     if (isVarParam) { mods :+= atPos(in.tokenPos, in.tokenPos)(Mod.VarParam()); next() }
-    val name = termName()
-    val tpt = {
-      accept[`:`]
-      val tpt = paramType()
-      if (tpt.isInstanceOf[Type.Arg.ByName]) {
-        def mayNotBeByName(subj: String) =
-          syntaxError(s"$subj parameters may not be call-by-name", at = name)
-        val isLocalToThis: Boolean = {
-          val isExplicitlyLocal = mods.accessBoundary.map(_.isInstanceOf[Term.This]).getOrElse(false)
-          if (ownerIsCase) isExplicitlyLocal
-          else isExplicitlyLocal || (!isValParam && !isVarParam)
+    val name = termName() match {
+      case q: Term.Name.Quasi => q.become[Term.Param.Name.Quasi]
+      case x => x
+    }
+    val tpt =
+      if (token.isNot[`:`])
+        None
```

Comment:
This should be handled elsewhere, because putting this logic in `param` means that previously invalid code such as `def foo(x) = ???` will now be parsed successfully.